src/main/java/mate/academy/spring/dao/UserDaoImpl.java

```diff
@@ -42,7 +43,16 @@ public void add(User user) {
             return session.createQuery("FROM User", User.class)
                     .getResultList();
         } catch (Exception e) {
-            throw new RuntimeException("User not found", e);
+            throw new RuntimeException("Users not found", e);
+        }
+    }
+
+    @Override
+    public Optional<User> get(Long id) {
+        try (Session session = sessionFactory.openSession()) {
+            return Optional.ofNullable(session.get(User.class, id));
+        } catch (Exception e) {
+            throw new RuntimeException("User not found " + id, e);
```

Comment:
If we have exception here it means smt went wrong with request execution even if user may be there